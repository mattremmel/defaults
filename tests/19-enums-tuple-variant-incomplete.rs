use defaults::Defaults;

#[derive(Defaults)]
#[default(value = "A")]
enum Foo {
    A(usize),
    B,
}

fn main() {}
