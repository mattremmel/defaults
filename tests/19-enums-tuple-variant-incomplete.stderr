error[E0308]: mismatched types
 --> $DIR/19-enums-tuple-variant-incomplete.rs:4:19
  |
4 | #[default(value = "A")]
  |                   ^^^
  |                   |
  |                   expected enum `Foo`, found fn item
  |                   expected `Foo` because of return type
5 | enum Foo {
6 |     A(usize),
  |     -------- fn(usize) -> Foo {Foo::A} defined here
  |
  = note: expected enum `Foo`
          found fn item `fn(usize) -> Foo {Foo::A}`
help: use parentheses to instantiate this tuple variant
  |
4 | #[default(value = "A"(_))]
  |                      ^^^
